src/generators/json/utils/sections/index.mjs

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+import { createSectionBase } from './base.mjs';
+import { createClassSection } from './class.mjs';
+import { createEventSection } from './event.mjs';
+import { createMethodSection } from './method.mjs';
+import { createModuleSection } from './module.mjs';
+import { createPropertySection } from './property.mjs';
+import { BASE_URL } from '../../../../constants.mjs';
+import { buildHierarchy } from '../../../../utils/buildHierarchy.mjs';
+import { GeneratorError } from '../../../../utils/generator-error.mjs';
+import { SCHEMA_FILENAME } from '../../constants.mjs';
+
+/**
+ * Processes children of a given entry and updates the section.
+ * @param {import('../../../../utils/buildHierarchy.mjs').HierarchizedEntry} entry - The current entry.
+ * @param {import('../../types.d.ts').Section | undefined} parent
+ */
+const handleChildren = ({ hierarchyChildren }, parent) =>
+  hierarchyChildren?.forEach(child => createSectionProperties(child, parent));
+
+/**
+ * @param {import('../../../../utils/buildHierarchy.mjs').HierarchizedEntry} entry
+ * @param {import('../../types.d.ts').Section | undefined} parent
+ * @param {string} version
+ * @returns {import('../../types.d.ts').Section}
+ */
+function createSectionProperties(entry, parent, version) {
+  /**
+   * @type {import('../../types.d.ts').Section}
+   */
+  let section;
+  try {
+    section = createSectionBase(entry, parent?.type);
+
+    // Temporarily add the parent section to the section so we have access to
+    //  it and can easily traverse through them when we need to
+    section.parent = parent;
```

Comment:
Nitpick: I don't _love_ adding an deleting a property, but we can always iterate on the implementation at a future time